### ISSUE TITLE
Exception while creating PDF file from Asciidoctor

### ISSUE THREAD

**Issue by kkgv** [asciidoctor/asciidoctorj#532]
I am getting exception while creating pdf file from ascii doc files using the following libraries in Spring boot application.
        
    org.asciidoctor:asciidoctorj:1.6.0-alpha.3
    org.asciidoctor:asciidoctorj-pdf:1.5.0-alpha.11
    org.asciidoctor:asciidoctorj-epub3:1.5.0-alpha.6
             
I found similar issues reported earlier tried by setting class loader but no success.

The exception shown is 

   ```
 org.jruby.exceptions.RaiseException: (LoadError) no such file to load -- asciidoctor
            at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:944) ~[jruby-core-9.1.2.0.jar!/:?]
            at RUBY.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:55) ~[?:?]
            at RUBY.<top>(<script>:9) ~[?:?]
```

My code looks like this. The path is the directory where ascii doc are stored. 

```
    final Asciidoctor asciidoctor = create();
    asciidoctor.renderDirectory(new AsciiDocDirectoryWalker(path), options().backend("pdf").get());
    asciidoctor.shutdown();

```

**Comment by robertpanzer** [comment]
Just tried this example in a plain Java program and it worked fine.
I have no clue how class loading works in a Spring Boot application.

But the error seems to be that JRuby can't find the Asciidoctor Ruby files which are located in `/gems` in the asciidoctor-1.6.0-alpha.3.jar.

**Comment by kkgv** [comment]
I tried setting class loader and gem path. Still getting same error. Here is the code.
`final RubyInstanceConfig configuration = new RubyInstanceConfig();
        configuration.setLoader(this.getClass().getClassLoader());
        final Asciidoctor asciidoctor = create(this.getClass().getClassLoader(), "gems/asciidoctorj-1.5.4/lib");
        asciidoctor.renderDirectory(new AsciiDocDirectoryWalker(path), options().backend("pdf").get());
        asciidoctor.shutdown();`

**Comment by abelsromero** [comment]
I could reproduce it.
The issue only happens when executing from a fatJar, and I guess is related to the custom classloader of SpringBoot. I remember they have a custom packaging system so the fat jar contains the dependency jars inside.
Maybe building the a nromal fat Jar with shadow plugin could fix it?

**Comment by kkgv** [comment]
Tried by building spring boot application as a war but it is still getting the same error.

**Comment by kkgv** [comment]
This issue resolved please refer #7399.

**Comment by robertpanzer** [comment]
Thank you for linking to a solution!
